File: BrewPOS-Workspace/brewpos-backend/src/routes/category.ts
import { Router } from 'express';
import { prisma } from '../db';

const router = Router();

// Get all categories
router.get('/', async (req, res) => {
  try {
    const categories = await prisma.category.findMany();
    res.json(categories);
  } catch (error) {
    console.error('Error fetching categories:', error);
    res.status(500).json({ error: 'Failed to fetch categories', details: error instanceof Error ? error.message : String(error) });
  }
});

// Create category
router.post('/', async (req, res) => {
  try {
    const { name } = req.body;
    const category = await prisma.category.create({
      data: { name },
    });
    res.status(201).json(category);
  } catch (error) {
    console.error('Error creating category:', error);
    res.status(500).json({ error: 'Failed to create category', details: error instanceof Error ? error.message : String(error) });
  }
});

export default router;

File: BrewPOS-Workspace/brewpos-flask/app/routes/category.py
from flask import Blueprint, request, jsonify
from app.extensions import db
from app.models.category import Category

category_bp = Blueprint('category', __name__, url_prefix='/api/categories')

@category_bp.route('', methods=['GET'])
def get_categories():
    try:
        categories = Category.query.all()
        return jsonify([c.to_dict() for c in categories])
    except Exception as e:
        print(f"Error fetching categories: {e}")
        return jsonify({'error': 'Failed to fetch categories', 'details': str(e)}), 500

@category_bp.route('', methods=['POST'])
def create_category():
    data = request.get_json() or {}
    name = data.get('name')
    if not name:
        return jsonify({'error': 'Name is required'}), 400

    try:
        category = Category(name=name)
        db.session.add(category)
        db.session.commit()
        return jsonify(category.to_dict()), 201
    except Exception as e:
        db.session.rollback()
        print(f"Error creating category: {e}")
        return jsonify({'error': 'Failed to create category', 'details': str(e)}), 500
